src/util/validation.cpp

```diff
@@ -5,15 +5,4 @@
 
 #include <util/validation.h>
 
-#include <consensus/validation.h>
-#include <tinyformat.h>
-
-/** Convert ValidationState to a human-readable message for logging */
-std::string FormatStateMessage(const ValidationState &state)
-{
-    return strprintf("%s%s",
-        state.GetRejectReason(),
-        state.GetDebugMessage().empty() ? "" : ", "+state.GetDebugMessage());
-}
-
 const std::string strMessageMagic = "Bitcoin Signed Message:\n";
```

Comment:
It seems a bit sad to leave this const all by itself in this implementation file. Perhaps move it to the header file and change to a constexpr? Bonus points for renaming `util/validation.h` to `util/strings.h` since it isn't anything to do with validation any more, and I can't find anywhere that would be a better home for this const.

Reply:
I don't thing `std::string` can be constexpr. Also, moving it to the header file might inline it in every translation unit. Probably doesn't matter for a short string like this, but no need to change existing code to make it worse.

Reply:
You're right. `std::string` can't be constexpr.

Reply:
Looking at the references to `strMessageMagic`, they are always of the form:

```c++
CHashWriter ss(SER_GETHASH, 0);
ss << strMessageMagic;
ss << strMessage;
```

And then the result of its `GetHash` is passed to either `SignCompact` or `RecoverCompact`, and the variable `ss` is never used again.

So the above code could be refactored into a function where `strMessageMagic` is just an implementation detail. Given that, is there a more appropriate place for it? What does `ss` stand for here?

Reply:
> What does ss stand for here?

Not sure. Serialization stream?

Reply:
(there have been `CDataStream ss` variables in the codebase since Satoshi era)

Reply:
Perhaps I could move that boilerplate code to `util/hash.h` as `HashMessage`?

```c++
uint256 HashMessage(const std::string& message);
```

Open to more suitable naming for the file and function. Note that `util/string.h` already exists for string joining, so the plural version may be confusing.

Would likely leave this to a follow-up PR given I'd be refactoring call sites.

Reply:
> Open to more suitable naming for the file and function

Perhaps `/util/message.h`?

> Would likely leave this to a follow-up PR given I'd be refactoring call sites.

Yeah, anything other than a simple rename should be left as a follow-up PR.

Reply:
Opened #17557 with the refactor.

Reply:
Rebased and was able to remove `util/validation.h` and `util/validation.cpp` now that #17577 is merged.